csi/deployment_util.go

```diff
@@ -137,6 +138,34 @@ func getCommonDeployment(commonName, namespace, serviceAccount, image, rootDir s
 	return commonDeploymentSpec
 }
 
+func getAffinity(name string) *v1.Affinity {
+	labelSelector := &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "app",
+				Operator: metav1.LabelSelectorOpIn,
+				Values: []string{
+					name,
+				},
+			},
+		},
+	}
+
+	return &v1.Affinity{
+		PodAntiAffinity: &v1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
+				{
+					Weight: 1,
```

Comment:
Not sure should we configure the weight or not.

Reply:
> nit: I probably will go without the function getAffinity which makes it easier to read the manifest.

There are a lots of brackets after merging into the manifest.
If it is acceptable in our code convention, I can remove getAffinity().

> Not sure should we configure the weight or not.

There is only one affinity rule here, so removing it and using default value are maybe okay as well.

Reply:
Resend the patch followed @jenting's suggestions.

Reply:
>>Not sure should we configure the weight or not.

>There is only one affinity rule here, so removing it and using default value are maybe okay as well.

Although the k8s doc said weight is not required, but the longhon driver deploy crashes if the weight is not provided.

Reply:
Any reason for the crash?

Reply:
@innobead 
log error message: `spec.template.spec.affinity.podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution[0].weight: must be in the range 1-100`

Unfortunately, the default value set by kubernetes  is 0 which is not accepted by itself.